src/cysignals/implementation.c

```diff
@@ -528,3 +535,361 @@ static void sigdie(int sig, const char* s)
  * signals.pyx. These require some of the above functions, therefore
  * this include must come at the end of this file. */
 #include "macros.h"
+
+#else /* Windows version */
```

Comment:
I feel like this should probably be outright in a separate file.  It will be very difficult to maintain separately otherwise.

Reply:
> I feel like this should probably be outright in a separate file.

No. The code for POSIX and Windows is actually pretty similar. You should just use macros or `#ifdef` in the places where the implementation is different.

Reply:
That would be fine too as long as there's enough overlap.  There are some areas where it might be nice to define some macros.